include/osmium/thread/pool.hpp

```diff
@@ -162,7 +162,11 @@ namespace osmium {
 
             static constexpr int default_num_threads = 0;
 
-            static Pool& instance() {
+            explicit Pool(int num_threads = default_num_threads) :
```

Comment:
Having a forwarding constructor here doesn't buy us anything. Just move the existing constructor down here so it becomes public and make the second parameter default to `detail::get_work_queue_size()`.